--- a/kernel/sensors/t31/Kbuild	2024-01-15 13:08:46.859149669 +0300
+++ b/kernel/sensors/t31/Kbuild	2024-01-15 13:13:37.850203251 +0300
@@ -12,7 +12,7 @@
 include $(src)/sensors/$(SOC)/os03b10/Kbuild
 include $(src)/sensors/$(SOC)/sc200ai/Kbuild
 include $(src)/sensors/$(SOC)/sc2232h/Kbuild
-#include $(src)/sensors/$(SOC)/sc2332/Kbuild
+include $(src)/sensors/$(SOC)/sc2332/Kbuild
 include $(src)/sensors/$(SOC)/sc2335/Kbuild
 include $(src)/sensors/$(SOC)/sc2336/Kbuild
 include $(src)/sensors/$(SOC)/sc3338/Kbuild
